modules/templates/templates/template-detail.tsx

```diff
@@ -0,0 +1,327 @@
+import { GlobalCore } from "@/core/module/module.types";
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import {
+  GenesisTemplate,
+  GenesisTemplateField,
+} from "@/core/module/core.types";
+import { useTranslation } from "react-i18next";
+import styles from "./styles/template-detail.module.css";
+import { FaEdit, FaSave, FaTrash, FaArrowLeft, FaPlus } from "react-icons/fa";
+import MessageHandler from "@/core/message-handler";
+import DeleteConfirmation from "@/resources/containers/delete-confirmation";
+import Table from "@/resources/table/table";
+import FieldModal from "@/resources/containers/field-modal";
+import { useService } from "@/core/module/service.factory";
+
+const messageHandler = MessageHandler.get();
+type TableDataType = GenesisTemplateField & { key: string; name: string };
+
+const TemplateDetail = () => {
+  const router = useRouter();
+  const { t } = useTranslation();
+  const templateService = useService("templates");
```

Comment:
I think I changed userservice to use `Service.require` instead of `Service.get` so the result is always defined.
Not sure is merged here yet.